src/v/cluster/rm_stm.cc

```diff
@@ -896,6 +903,17 @@ void rm_stm::set_seq(model::batch_identity bid, model::offset last_offset) {
     }
 }
 
+void rm_stm::reset_seq(model::batch_identity bid) {
+    _log_state.seq_table.erase(bid.pid);
+    auto& seq = _log_state.seq_table[bid.pid];
```

Comment:
this requires two look ups. instead, you could construct an instance the way you want and then assign using [] access and reduce it to a single index lookup.